mod-source-record-manager-server/src/main/java/org/folio/services/afterprocessing/InstanceProcessingServiceImpl.java

```diff
@@ -55,14 +49,22 @@ public InstanceProcessingServiceImpl(@Autowired JobExecutionSourceChunkDao jobEx
       if (ar.failed()) {
         updateSourceChunkState(sourceChunkId, JobExecutionSourceChunk.State.ERROR, params);
       } else {
-        addAdditionalFields(recordToInstanceList, params);
+        List<Instance> result = ar.result();
+        List<Pair<Record, Instance>> recordsToUpdate = calculateRecordsToUpdate(recordToInstanceList, result);
+        addAdditionalFields(recordsToUpdate, params);
       }
       // Complete future in order to continue the import process regardless of the result of creating Instances
       future.complete();
     });
     return future;
   }
 
+  private List<Pair<Record, Instance>> calculateRecordsToUpdate(List<Pair<Record, Instance>> recordToInstanceList, List<Instance> result) {
+    return recordToInstanceList.stream()
+      .filter(it -> result.contains(it.getValue()))
```

Comment:
contains works with equals. Is there working fine? We can use only id field to understand that instance is the same